lib/src/aggregate_sample.dart

```diff
@@ -135,8 +135,6 @@ extension AggregateSample<T> on Stream<T> {
         } else {
           triggerSub!.pause();
```

Comment:
This pause happens only if the current stream is broadcast, and the `trigger` stream is not.

If this stream is a broadcast stream, then its emissions shouldn't depend on whether it has subscribers or not,
and I'd *not* pause the `triggerSub` here. Or `valueSub` for that matter.
There should be no difference for someone who listens, cancels and listens again whethere there are other listeners at the same time.


If this stream is not broadcast, then nothing ever needs to happen after cancel, and we should cancel `triggerSub` too.

So I'd make this:
```dart
controler.onCancel = () {
  if (!isBroadcast) {
    if (isValueDone) {
      if (isTriggerDone) return null;
      return triggerSub.cancel();
    } 
    if (isTriggerDone) return triggerSub!.cancel();
    return [valueSub!.cancel(), triggerSub!.cancel()].wait.then((_) => null);
  }
};
```
(Probably no need to set the `...Sub` variables to `null`. Nothing holds on to a cancelled completer.)

Also noticed: `onEmpty` should probably not be required if `longPoll: true` makes it not be used.
Making it optional, and then not emitting anything on a `longPoll:false` empty trigger would be consistent. You get a value when you ask for one, if there is one.

Or make `longPoll` the behavior when `onEmpty == null;`, instead of asking for it and then also passing an unusable `onEmpty`.

Reply:
(Just for the heck of it, here is a complete rewrite: https://dartpad.dev/?id=c834bfd326ab1341903145bb9e32bfa7
It moves almost everything inside `onListen`, which avoids some nullable types.)

Reply:
It doesn't pass the tests as-is, but there could be tests which are overly coupled to the particular async event interleaving. Maybe file this as an issue to consider?